### PR TITLE
docs: merge nix code into toplevel

### PR DESCRIPTION
**Pull request by flokli**
docs: merge nix code into toplevel
    
    For now, this introduces a second nixpkgs pin to the toplevel
    nix/sources.json, and exposes the doc environment as a `docsEnv`
    attribute.
    
    The binaries in there are added to $PATH by direnv.
    
    As Dockerfile can't copy files from outside its folder, and the only
    purpose of it was to provide a binaries to build as well (which we can
    just provide via Nix), it's removed.
    
    There's now an additional `.envrc` in `docs/`, adding doc-related
    tooling to $PATH whenever the `docs` folder is entered.
    
    Building can happen via a a top-level `docs` attribute, without the need
    for intermediate Docker containers.
    
    This removes the necessity for a bunch of Makefile rules and
    intermediate container images, as well as additional CI recipies - we
    can just build the nix attribute from GH Actions.
    
    The GH actions pipeline is extended to upload/update docs in our docs
    bucket. Permissions for this come via OIDC and the GitHub actions
    tokens.
    
    The documentation is updated to match the new state of things.
    
    Markdown rendering is updated from `recommonmark` to `myst-parser`.

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] If this PR changes development workflow or dependencies, they have been A) automated and B) documented under docs/developer/. All efforts have been taken to minimize development setup breakage or slowdown for co-workers.
 - [ ] If HTTP endpoint paths have been added or renamed, or feature configs have changed, the **endpoint / config-flag checklist** (see Wire-employee only backend [wiki page](https://github.com/zinfra/backend-wiki/wiki/Checklists)) has been followed.
 - [ ] If a cassandra schema migration has been added, I ran **`make git-add-cassandra-schema`** to update the cassandra schema documentation.
 - [ ] **changelog.d** contains the following bits of information ([details](https://github.com/wireapp/wire-server/blob/develop/docs/developer/changelog.md)):
   - [ ] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.
   - [ ] If new config options introduced: added usage description under docs/reference/config-options.md
   - [ ] If new config options introduced: recommended measures to be taken by on-premise instance operators.
   - [ ] If a cassandra schema migration is backwards incompatible (see also [these docs](https://github.com/wireapp/wire-server/blob/develop/docs/developer/cassandra-interaction.md#cassandra-schema-migrations)), measures to be taken by on-premise instance operators are explained.
   - [ ] If a data migration (not schema migration) introduced: measures to be taken by on-premise instance operators.
   - [ ] If public end-points have been changed or added: does nginz need un upgrade?
   - [ ] If internal end-points have been added or changed: which services have to be deployed in a specific order?

### CODE REVIEW

**Comment by flokli** [comment]
Right now, the PDF I get back from a `nix-build nix -A docs` looks very different from what's hosted at https://docs.wire.com/main.pdf - I didn't yet figure out why.

@arthurwolf, any pointers?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -28,3 +28,12 @@ jobs:
...
        run: nix-env -if ./nix -A devEnv
+     - name: Build docs
+       run: nix-build --no-out-link ./nix -A docs
+     - name: Deploy docs
+       if: github.ref == 'refs/heads/master'
+       run: |
+         cd docs
+         docs=$(nix-build --no-out-link ./nix -A docs)
+         aws s3 sync $docs/html s3://origin-docs.wire.com/
+         aws s3 cp $docs/pdf/wire_federation.pdf s3://origin-docs.wire.com/main.pdf
```
Comment (flokli) [inline .github/workflows/ci.yml:39]: We probably still needs secrets to successfully deploy, but let's do this once we agree on the rest of the PR.

**Comment by arthurwolf** [comment]
@flokli I have never touched the PDF part of the docs. I plan on looking into it, but can't just right now.

**Comment by flokli** [comment]
I updated the pipeline and the AWS IAM bits underneath, so this now has permission to properly update docs in the S3 bucket, once the PR hits develop.

> @flokli I have never touched the PDF part of the docs. I plan on looking into it, but can't just right now.

Well, the PDF still gets built, and CI verifies that - I think it just applies more of the styling/formatting we also use for the HTML version.

**Review by arthurwolf — APPROVED** [review]
tested, builds/works fine

**Pull request merged**